vmdb/app/models/service_template.rb

```diff
@@ -115,31 +116,37 @@ def type_display
   end
 
   def create_tasks_for_service(service_task, parent_svc)
-    tasks = []
     svc = create_service(service_task, parent_svc)
 
     user = User.find_by_userid(service_task.userid)
-    self.set_ownership(svc, user) unless user.nil?
+    set_ownership(svc, user) unless user.nil?
 
     service_task.destination = svc
-    self.service_resources.each do |child_svc_rsc|
+
+    create_subtasks(service_task, svc)
+  end
+
+  # default implementation to create subtasks from service resources
+  def create_subtasks(parent_service_task, parent_service)
+    tasks = []
+    service_resources.each do |child_svc_rsc|
       scaling_min = child_svc_rsc.scaling_min
       1.upto(scaling_min).each do |scaling_idx|
-        nh = service_task.attributes.dup
+        nh = parent_service_task.attributes.dup
```

Comment:
You did not write this, so no need to change. But thought I'd put in my 2 cents here:
```
  nh = parent_service_task.attribtues.except(created_on updated_on type state status message)
  hn['options'] = parent_service_task.options.except(:child_tasks)
```